tests/test_verification.py

```diff
@@ -172,3 +172,79 @@ def test_contradiction(self) -> None:
             end Foo;
             """,
             r'^contradicting condition 0 from field "F1" to "F2"')
+
+    def test_invalid_type_condition_range_low(self) -> None:
+        self.assert_parse_exception_string(
+            """
+            package Foo is
+                type Element is range 100..1000 with Size => 32;
+                type Bar is
+                    message
+                        F1 : Element then F2 if F1 < 100;
+                        F2 : Element;
+                    end message;
+            end Foo;
+            """,
+            r'^unreachable field "F2"')
+
+    def test_invalid_type_condition_range_high(self) -> None:
+        self.assert_parse_exception_string(
+            """
+            package Foo is
+                type Element is range 5..50000 with Size => 32;
+                type Bar is
+                    message
+                        F1 : Element then F2 if F1 > 60000;
+                        F2 : Element;
+                    end message;
+            end Foo;
+            """,
+            r'^unreachable field "F2"')
+
+    def test_invalid_type_condition_modular(self) -> None:
+        self.assert_parse_exception_string(
+            """
+            package Foo is
+                type Element is mod 2**8;
+                type Bar is
+                    message
+                        F1 : Element then F2 if F1 > 1000;
+                        F2 : Element;
+                    end message;
+            end Foo;
+            """,
+            r'^unreachable field "F2"')
+
+    def disabled_test_invalid_type_condition_enum(self) -> None:
```

Comment:
Please always add a reference to an issue in the form `# ISSUE: Componolit/RecordFlux#XX` above a disabled test.

Reply:
Addressed in e03aa1e